core/embed/bootloader_ci/messages.c

```diff
@@ -402,8 +402,6 @@ static bool _read_payload(pb_istream_t *stream, const pb_field_t *field,
   return true;
 }
 
-secbool check_vendor_header_keys(const vendor_header *const vhdr);
-
 static int version_compare(uint32_t vera, uint32_t verb) {
```

Comment:
I'm not entirely sure if this can or can't overflow/underflow due to integer promotion rules and so on. I.e. if you have highest bit set in `vera` or `verb`, is it going to behave correctly?
I see warning that narrowing `uint32_t` into `int` this way is implementation-defined, so compiler can theoretically change behavior even if it behaves correctly now in some optimization in later version. Someone should double-check that we are not depending on undefined/implementation-specific behavior.
I suspect that this might promote the `uint32_t` into `int`, and shift-right on `int` and right shift on negative integer is implementation-defined. Probably a check that makes it error if either `vera` or `verb` have highest bit set.